test-e2e/sync.js

```diff
@@ -49,47 +52,59 @@ test('Create and sync data', { timeout: 100_000 }, async function (t) {
   connectPeers(managers, { discovery: false })
   await waitForSync(projects, 'initial')
 
-  for (const schemaName of generatedSchemaNames) {
-    for (const project of projects) {
-      const deviceId = project.deviceId.slice(0, 7)
-      // @ts-ignore - to complex to narrow `schemaName` to valid values
-      const docs = await project[schemaName].getMany()
-      const expected = generatedDocs.filter((v) => v.schemaName === schemaName)
-      if (SCHEMAS_INITIAL_SYNC.includes(schemaName)) {
-        t.alike(
-          sortById(docs),
-          sortById(expected),
-          `All ${schemaName} docs synced to ${deviceId}`
-        )
-      } else {
-        t.not(
-          docs.length,
-          expected.length,
-          `Not all ${schemaName} docs synced to ${deviceId}`
-        )
-      }
-    }
-  }
+  await Promise.all(
+    map(generatedSchemaNames, (schemaName) =>
+      Promise.all(
+        map(projects, async (project) => {
+          const deviceId = project.deviceId.slice(0, 7)
+          // @ts-ignore - to complex to narrow `schemaName` to valid values
+          const docs = await project[schemaName].getMany()
+          const expected = generatedDocs.filter(
+            (v) => v.schemaName === schemaName
+          )
+          if (SCHEMAS_INITIAL_SYNC.includes(schemaName)) {
+            assert.deepEqual(
+              new Set(docs),
+              new Set(expected),
+              `All ${schemaName} docs synced to ${deviceId}`
+            )
+          } else {
+            assert.notEqual(
+              docs.length,
+              expected.length,
+              `Not all ${schemaName} docs synced to ${deviceId}`
+            )
+          }
+        })
+      )
+    )
+  )
 
   for (const project of projects) {
     project.$sync.start()
   }
 
   await waitForSync(projects, 'full')
 
-  for (const schemaName of generatedSchemaNames) {
-    for (const project of projects) {
-      const deviceId = project.deviceId.slice(0, 7)
-      // @ts-ignore - to complex to narrow `schemaName` to valid values
-      const docs = await project[schemaName].getMany()
-      const expected = generatedDocs.filter((v) => v.schemaName === schemaName)
-      t.alike(
-        sortById(docs),
-        sortById(expected),
-        `All ${schemaName} docs synced to ${deviceId}`
+  await Promise.all(
+    map(generatedSchemaNames, (schemaName) =>
+      Promise.all(
+        map(projects, async (project) => {
+          const deviceId = project.deviceId.slice(0, 7)
+          // @ts-ignore - to complex to narrow `schemaName` to valid values
```

Comment:
nit: may as well fix the grammar 😄 

```suggestion
          // @ts-ignore - too complex to narrow `schemaName` to valid values
```

Reply:
Done in da8dba059dac0a9263de73f352ceca748aff2efe.